gammapy/maps/wcs.py

```diff
@@ -882,6 +882,38 @@ def __repr__(self):
         )
         return str_
 
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            # check overall shape and axes compatibility
+            if self.data_shape != other.data_shape:
+                return False
+
+            result = True
+            for axis, otheraxis in zip(self.axes, other.axes):
+                result &= axis == otheraxis
+            # check WCS consistency
+            return result and self.wcs.wcs.compare(other.wcs.wcs, tolerance=1e-6)
+        return NotImplemented
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def check_equal(self, other):
```

Comment:
Looking at this PR now, I wonder if the addition of `check_equal` with these two lines in addition to `__eq__` is really useful. I mean, the difference for callers is:
```
geom.check_equal(geom2)
```
vs
```
if geom != geom2:
    raise ValueError()
```
so one line more, and the caller has full control over what to raise.

I'm +1 to remove the `check_equal`, which will save ~ 100 lines of tests and boilerplate code we'll have to maintain.
(and for consistency we should otherwise probably add `check_equal` to many other Gammapy objects).

Sorry to only state this opinion now, it wasn't clear to me when looking at this PR last time.

Reply:
OK I agree but the tests have to remain to test the `__eq__` methods.

Reply:
Yes, of course the tests should remain. But they will simplify to just `assert a == b` or `assert a != b` instead of the current `with pytest.raises` bonanza, no?